utopian-io/v2.utopian.io
# Add user contribution rights check in the project view page

I've renamed the get method in the api because it will be only used to get the information for editing purposes.
The method to load data to display information is handled by another where the contents of the selected tab are also loaded

### Verify that
Git checklist

Front end checklist

API checklist

Tests and comments checklist

Please provide a test plan so that the reviewers can see how you tested your work and/or the different acceptance criteria described in the task.

### **Test plan**

_External collaborators disabled_

* Create a project disabling external contribution
* Connect to the app with a user that is not an owner or a collaborator

**Expected result:** You should see the buttons disabled and a tooltip message

_External collaborator enabled_

* Create a project enabling external contribution
* Connect to the app with a user that is not an owner or a collaborator

**Expected result:** You should see the buttons in a normal state

_External collaborators disabled but you're the owner_

* Create a project disabling external contribution
* Connect to the app with a user that is an owner

**Expected result:** You should see the buttons in a normal state

_External collaborators disabled but you're a collaborator with the articles role_

* Create a project disabling external contribution
* Add a user with the articles roles
* Connect to the app with this user

**Expected result:** You should see the button article in a normal state but the bounty disabled with a tooltip saying that he doesn't have the rights

_External collaborators disabled but you're a collaborator with the bounties role_

* Create a project disabling external contribution
* Add a user with the bounties role
* Connect to the app with this user

**Expected result:** You should see the button bounties in a normal state but the article disabled with a tooltip saying that he doesn't have the rights

_External collaborators disabled but you're a collaborator with all the roles_

* Create a project disabling external contribution
* Add a user with the articles and bounties  roles
* Connect to the app with this user

**Expected result:** You should see the buttons in a normal state
